client/ui/controllers/settingsController.cpp

```diff
@@ -140,9 +140,18 @@ void SettingsController::restoreAppConfigFromData(const QByteArray &data)
 {
     bool ok = m_settings->restoreAppConfig(data);
     if (ok) {
+        QJsonObject new_config_data = QJsonDocument::fromJson(data).object();
+
         m_serversModel->resetModel();
         m_languageModel->changeLanguage(
                 static_cast<LanguageSettings::AvailableLanguageEnum>(m_languageModel->getCurrentLanguageIndex()));
+
+#if defined(Q_OS_WINDOWS) || defined(Q_OS_ANDROID)
+        bool appSplittunnelingEnabled = new_config_data.value("Conf/appsSplitTunnelingEnabled").toBool();
+        m_appSplitTunnelingModel->toggleSplitTunneling(appSplittunnelingEnabled);
```

Comment:
u should also call the setRouteMode() function for both site and app split tunneling